### PR TITLE
Rearranging the public/protected/private chunks of classes

### PR DESCRIPTION
**Pull request by jamiebramwell**
At some point, I got in the bad habit of putting private class members above public ones. This PR goes toward fixing that.

### CODE REVIEW

File: src/solvers/nonlinear_solid_operators.hpp [hunk 1]
```diff
@@ -114,56 +158,53 @@ class NonlinearSolidReducedSystemOperator : public mfem::Operator {
...
+class NonlinearSolidDynamicOperator : public mfem::TimeDependentOperator {
 public:
   /**
-   * @brief Construct a new Nonlinear Solid Reduced System Operator object
+   * @brief Construct a new Nonlinear Solid Dynamic Operator object
    *
    * @param[in] H_form The nonlinear stiffness form
    * @param[in] S_form The linear viscosity form
-   * @param[in] M_form The linear mass form
+   * @param[in] M_form The lineawr mass form
```
Comment (joshessman-llnl) [inline src/solvers/nonlinear_solid_operators.hpp]: linear

**Review by joshessman-llnl — APPROVED** [review]

**Comment by jamiebramwell** [comment]
LGTM

File: src/solvers/nonlinear_solid_operators.hpp [hunk 2]
```diff
@@ -74,6 +74,50 @@ class NonlinearSolidQuasiStaticOperator : public mfem::Operator {
...
+  /**
+   * @brief Construct a new Nonlinear Solid Reduced System Operator object
+   *
+   * @param[in] H_form The nonlinear stiffness form
+   * @param[in] S_form The linear viscosity form
+   * @param[in] M_form The linear mass form
+   * @param[in] ess_bdr The essential boundary conditions
+   */
+  NonlinearSolidReducedSystemOperator(const mfem::ParNonlinearForm& H_form, const mfem::ParBilinearForm& S_form,
+                                      mfem::ParBilinearForm&                       M_form,
```
Comment (samuelpmishLLNL) [inline src/solvers/nonlinear_solid_operators.hpp:87]: const?

File: src/solvers/nonlinear_solid_operators.hpp [hunk 3]
```diff
@@ -114,56 +158,53 @@ class NonlinearSolidReducedSystemOperator : public mfem::Operator {
...
-   * @param[in] x The current position
+   * @param[in] vx The current velocity and displacement state
+   * @param[out] dvx_dt The explicit time derivative of the state vector
    */
-  void SetParameters(double dt, const mfem::Vector* v, const mfem::Vector* x);
+  virtual void Mult(const mfem::Vector& vx, mfem::Vector& dvx_dt) const;
 
   /**
-   * @brief Compute y = H(x + dt (v + dt k)) + M k + S (v + dt k).
+   * @brief Solve the Backward-Euler equation: k = f(x + dt*k, t), for the unknown k.
```
Comment (samuelpmishLLNL) [inline src/solvers/nonlinear_solid_operators.hpp:193]: Information is being lost by changing these comments. This object isn't computing some arbitrary f(x,t), it's doing something specific, namely `//Compute y = H(x + dt (v + dt k)) + M k + S (v + dt k).`

File: src/solvers/nonlinear_solid_operators.hpp [hunk 4]
```diff
@@ -114,56 +158,53 @@ class NonlinearSolidReducedSystemOperator : public mfem::Operator {
...
    *
-   * @param[in] k The input state to evaluate the residual
-   * @param[out] y The output state to evalutae the residual
-   */
-  virtual void Mult(const mfem::Vector& k, mfem::Vector& y) const;
-
-  /**
-   * @brief Compute J = M + dt S + dt^2 grad_H(x + dt (v + dt k)).
```
Comment (samuelpmishLLNL) [inline src/solvers/nonlinear_solid_operators.hpp:149]: see above

File: src/solvers/nonlinear_solid_operators.hpp (hunk 2 above)
Comment (jamiebramwell) [reply src/solvers/nonlinear_solid_operators.hpp:87]: Unfortunately we have to call `ParallelAssemble` on the mass matrix, so this object is actually not const. The other forms just need to be evaluated in a matrix-free fashion.

**Review by white238 — APPROVED** [review]

**Comment by jamiebramwell** [comment]
LGTM

**Pull request merged**